### PR TITLE
Fix issues to match contributing guidance

### PR DESCRIPTION
**Pull request by sunbinzhu**
1. Fetch the scripts used in this template in the template folder rather than my own storage account.
2. Update the PrepareHN.ps1 to support UnSecure DNS dynamic update
3. Update some texts.

### CODE REVIEW

**Comment by azurecla** [comment]
Hi __@sunbinzhu__, I'm your friendly neighborhood Azure Pull Request Bot (You can call me AZPRBOT). Thanks for your contribution!
    <span>
        In order for us to evaluate and accept your PR, we ask that you sign a contribution license agreement. It's all electronic and will take just minutes. I promise there's no faxing. https://cla.azure.com.
    </span>

TTYL, AZPRBOT;

**Comment by azurecla** [comment]
Hi __@sunbinzhu__, I'm your friendly neighborhood Azure Pull Request Bot (You can call me AZPRBOT). Thanks for your contribution!
    <span>
        In order for us to evaluate and accept your PR, we ask that you sign a contribution license agreement. It's all electronic and will take just minutes. I promise there's no faxing. https://cla.azure.com.
    </span>

TTYL, AZPRBOT;

File: create-hpc-cluster/azuredeploy.json [hunk 1]
```diff
@@ -107,8 +106,8 @@
             }
         },
         "headNodePostConfigScript": {
-            "type": "string", 
-            "defaultValue": " ",           
+            "type": "string",
+            "defaultValue": " ",
```
Comment (singhkay) [inline create-hpc-cluster/azuredeploy.json]: What's the purpose of the string with a space as a default value?

File: create-hpc-cluster/azuredeploy.json [hunk 2]
```diff
@@ -121,25 +120,24 @@
...
         "publicIPAddressNameHN": "[concat(variables('publicIPAddressName'),'-HN')]",
         "dnsName": "[concat(parameters('clusterName'), 'ep')]",
         "nicName": "[concat(parameters('clusterName'),'-nic')]",
         "nicNameHN": "[concat(variables('nicName'),'-hn')]",
-        "vmExtensionName": "customScriptExtension",
         "availableCNImageSkus": {
             "ComputeNode": "2012R2CN",
             "ComputeNodeWithExcel": "2012R2CNExcel"
         },
         "computeNodeImageSku": "[variables('availableCNImageSkus')[parameters('computeNodeImage')]]",
```
Comment (singhkay) [inline create-hpc-cluster/azuredeploy.json]: Do you need a "." here to access the key within the object?

File: create-hpc-cluster/azuredeploy.json (hunk 1 above)
Comment (sunbinzhu) [inline create-hpc-cluster/azuredeploy.json]: This is legacy code, a space here represents no post config script.

File: create-hpc-cluster/azuredeploy.json (hunk 2 above)
Comment (sunbinzhu) [inline create-hpc-cluster/azuredeploy.json]: variables('availableCNImageSkus') is the dictionary, and parameters('computeNodeImage') is the key. I am using dictionary[key] to get the value.

**Comment by singhkay** [comment]
Looks good! merged

**Pull request merged**